tests/consumer.rs

```diff
@@ -436,3 +501,14 @@ where
         _ = tokio::time::sleep(Duration::from_secs(1)) => {},
     };
 }
+
+fn record_with_timestamp_milliseconds(key: &[u8], timestamp: DateTime<Utc>) -> Record {
+    Record {
+        key: Some(key.to_vec()),
+        value: Some(b"hello kafka".to_vec()),
+        headers: std::collections::BTreeMap::from([("foo".to_owned(), b"bar".to_vec())]),
+        timestamp: timestamp
+            .with_nanosecond(timestamp.nanosecond() / 1_000_000 * 1_000_000)
```

Comment:
what does that do?

Reply:
Tailor timestamp to millisecond. Since the time precision in Kafka is at millisecond level.